share/picongpu/tests/CurrentDeposition/lib/python/test/CurrentDeposition/MainTest.py

```diff
@@ -0,0 +1,191 @@
+import openpmd_api as io
+import numpy as np
+from grid_class import grid
+import sys
+
+
+def get_params(path):
+    """creates a series of the data found at path.
+    path should have the form data_%T to provide access to all iterations.
+
+    returns: "./simData_%T.h5"
+        series
+        order... order of assignement function
+        parameter list... list of parameters of the simulation (units etc.)"""
+    # load series
+    series = io.Series(path, io.Access.read_only)
+
+    if len(series.iterations) == 1:
+        print("There is just 1 iteration in the series")
+        print("make sure, there are at least two")
+        longer_series = False
+
+    else:
+        longer_series = True
+
+    # read order assignment function
+    i = series.iterations[0]
+    electrons = i.particles["e"]
+    order = int(electrons.get_attribute("particleShape"))
+
+    # read parameters of the simulation
+    cell_depth = i.get_attribute("cell_depth")
+    cell_height = i.get_attribute("cell_height")
+    cell_width = i.get_attribute("cell_width")
+    dt = i.get_attribute("dt")
+    unit_time = i.get_attribute("unit_time")
+    unit_length = i.get_attribute("unit_length")
+    unit_charge = i.get_attribute("unit_charge")
+
+    return longer_series, series, order, [
+        cell_depth, cell_height, cell_width, dt, unit_time, unit_charge,
+        unit_length
+        ]
+
+
+def read_series(series):
+    """reads the position values position_offset (grid_knot) and position
+    (float in the intervall [0,1), relativ position in the cell) and the
+    current density provided by the simulation as well the charge of the
+    particle.
+
+    returns:
+        Js... current density list (contains array of each component for each
+                                    iteration)
+        poss... array of all relative in-Cell positions
+        pos_offs... array of all knot-positions
+        charge.. charge of particle
+        """
+    num_iter = len(series.iterations)
+
+    # create arrays for saving the data
+    Js = []
+    poss = np.zeros((num_iter, 3))
+    pos_offs = np.zeros((num_iter, 3))
+
+    for iteration in range(0, num_iter, 1):
+        i = series.iterations[iteration]  # current position
```

Comment:
A `# current timestep` or `# current iteration` would be more specific.

Reply:
done